diff -Naur redis-2.8.13.orig/redis.conf redis-2.8.13/redis.conf
--- redis-2.8.13.orig/redis.conf	2014-07-14 10:48:14.000000000 -0500
+++ redis-2.8.13/redis.conf	2014-08-01 09:41:11.000000000 -0500
@@ -34,7 +34,7 @@
 
 # By default Redis does not run as a daemon. Use 'yes' if you need it.
 # Note that Redis will write a pid file in /var/run/redis.pid when daemonized.
-daemonize no
+daemonize yes
 
 # When running daemonized, Redis writes a pid file in /var/run/redis.pid by
 # default. You can specify a custom pid file location here.
